fuel-core-interfaces/src/txpool.rs

```diff
@@ -26,61 +26,66 @@ pub trait TxPoolDb:
     }
 }
 
-/// Subscriber interface that receive inserted/removed events from txpool.
-/// Usually needed for libp2p broadcasting and can be used to notify users of new tx.
-#[async_trait]
-pub trait Subscriber: Send + Sync {
-    async fn inserted(&self, tx: Arc<Transaction>);
-
-    async fn inserted_on_block_revert(&self, tx: Arc<Transaction>);
-
-    async fn removed(&self, tx: Arc<Transaction>, error: &Error);
-}
-
-#[async_trait]
-pub trait TxPool: Send + Sync {
+#[derive(Debug)]
+pub enum TxPoolMpsc {
+    /// Return all sorted transactions that are includable in next block.
+    /// This is going to be heavy operation, use it with only when needed.
+    Includable {
```

Comment:
[nit] should we create a tech debt item to explore parameterizing this response a bit? For example we could put an upper-bound on the number of txs to return.

Reply:
In most cases, you would like to fetch all transactions from txpool as you don't know how efficient you can pack the new block as it depends on gas that is spent by transaction. There could possible be some consensus wide limit calculated as `block_gas_limit` / `  system minimum of transaction gas spend` but those things are like a small detail.

Reply:
Agreed - this will come up later if needed